src/gpu/intel/jit/gemm/selector/db/kernel.db

```diff
@@ -1216,8 +1216,10 @@ auto _CATALOG_ = kcatalog::toArray({
-{{'G', "gemm", {"Q", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 33, -1}, {-1, 64, -1}, {-1, 33, -1}, {-1, 64, -1}, {16, 16, 1}, "IAB"}, "at32x2+m64@64 am64+m64@64 aB wg 4x4 xaf vav sr br sb64 bm0 bk0 sys sm sn grf256 st np", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 4, 1}, 1, (WGType) 1, 1, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
-{{'G', "gemm", {"Q", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 65, -1}, {-1, 128, -1}, {-1, 65, -1}, {-1, 128, -1}, {16, 16, 1}, "IAB"}, "at32x2+m64@64 am64+m64@64 aB wg 4x4 xaf vav sr br sb64 bm0 bk0 sys grf256 sm sn st np", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {32, 32, 64}, {4, 4, 1}, 1, (WGType) 1, 1, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
+{{'G', "gemm", {"Q", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 33, -1}, {-1, 48, -1}, {-1, 33, -1}, {-1, 48, -1}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x2 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 24, 64}, {4, 2, 1}, 1, (WGType) 1, 1, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
+{{'G', "gemm", {"Q", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 49, -1}, {-1, 64, -1}, {-1, 49, -1}, {-1, 64, -1}, {16, 16, 1}, "IAB"}, "at32x2+m64@64 am64+m64@64 aB wg 4x4 xaf vav sr br sb64 bm0 bk0 sys sm sn grf256 st np", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {32, 16, 64}, {4, 4, 1}, 1, (WGType) 1, 1, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
+{{'G', "gemm", {"Q", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 65, -1}, {-1, 96, -1}, {-1, 65, -1}, {-1, 96, -1}, {16, 16, 1}, "IAB"}, "at32x2+m64@64 am64+m64@64 aB wg 4x4 xaf vav sr br sb64 bm0 bk0 sys grf256 sm sn st np", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {32, 24, 64}, {4, 4, 1}, 1, (WGType) 1, 1, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
+{{'G', "gemm", {"Q", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 97, -1}, {-1, 128, -1}, {-1, 97, -1}, {-1, 128, -1}, {16, 16, 1}, "IAB"}, "at32x2+m64@64 am64+m64@64 aB wg 4x4 xaf vav sr br sb64 bm0 bk0 sys grf256 sm sn st np", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {32, 32, 64}, {4, 4, 1}, 1, (WGType) 1, 1, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
```

Comment:
For this workload, what `M` range corresponds with the transition from memory-bound to compute bound? In particular, the lack of any k-parallelization in these strategies suggest to me that they will have poor load balancing behavior. What range of shapes were these validated on?

Reply:
Just to demonstrate what I am seeing, I did a sweep of N values checking performance. The effect of poor load balancing is fairly significant until N  ~= 30,000 which is larger than most vLLM workloads. In my experience, we can normally find strategies with local k-parallelization of 2 for minimal overhead, which would significantly improve the load balancing. 

<img width="1324" height="739" alt="image" src="https://github.com/user-attachments/assets/c7fb5777-be16-4127-97ee-9c50242d216e" />